CMakeLists.txt

```diff
@@ -49,6 +52,33 @@ ELSE ()
     set(CMAKE_EXE_LINKER_FLAGS "-pthread ${CMAKE_EXE_LINKER_FLAGS}")
 ENDIF()
 
+if(ENABLE_W_ERROR)
+    set(CMAKE_CXX_FLAGS "-Werror ${CMAKE_CXX_FLAGS_DEBUG}")
```

Comment:
This probably should be
```CMake
set(CMAKE_CXX_FLAGS "-Werror ${CMAKE_CXX_FLAGS}") 
set(CMAKE_CXX_FLAGS_DEBUG "-Werror ${CMAKE_CXX_FLAGS_DEBUG}") 
```

Otherwise when setting the `ENABLE_W_ERROR` flag the DEBUG flags are always included.